src/Crate/PDO/PDOStatement.php

```diff
@@ -202,7 +202,10 @@ public function execute($input_parameters = null)
             if (is_int($parameter) && $zero_based) {
                 $parameter++;
             }
-            $this->bindValue($parameter, $value);
+            // to correctly save arrays and objects, we need to pass
```

Comment:
Dne 12. 05. 2015 ob 12:48 je Marco Pivetta zapisal(a):
> In src/Crate/PDO/PDOStatement.php
> <https://github.com/crate/crate-pdo/pull/21#discussion_r30122387>:
>
>> @@ -202,7 +202,10 @@ public function execute($input_parameters = null)
>>              if (is_int($parameter) && $zero_based) {
>>                  $parameter++;
>>              }
>> -            $this->bindValue($parameter, $value);
>> +            // to correctly save arrays and objects, we need to pass
>
> Not sure why you are automating this sort of operation. You are supposed
> to pass in |PDO::PARAM_ARRAY| or |PDO::PARAM_OBJECT| (manually) when
> binding parameters.

It's a patch for using execute() with input_parameters. When you send 
parameters with execute($params) and not when binding separately with 
bindValue().

Laravel (using Eloquent) creates a prepared statement like:
   INSERT INTO Users (id, name, arrField) values (?, ?, ?);

It then executes it with:
   $stmt->execute($arrayOfValues);

crate-pdo takes over from here and binds values to parameters:

https://github.com/crate/crate-pdo/blob/master/src/Crate/PDO/PDOStatement.php#L201

foreach ($input_parameters_array as $parameter => $value) {
     if (is_int($parameter) && $zero_based) {
         $parameter++;
     }
     $this->bindValue($parameter, $value);
                                        ^^^^^ missing type
}

This loop does the actual binding but it does it without data type. It 
assumes PDO::PARAM_STR as it is the default in bindValue():

public function bindValue($parameter, $value, $data_type = PDO::PARAM_STR)

So we bind all parameters as strings, which produces an error.

Ratko

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crate/crate-pdo/pull/21/files#r30122387>.
>